### ISSUE TITLE
ensure `exclude:` path filtering works in multi-option contexts

### ISSUE THREAD

**Issue by pq** [dart-lang/sdk#54858]
Looking at failures in https://dart-review.googlesource.com/c/sdk/+/350342, it appears that `excludes` aren't working properly.

For example:

```
error - _fe_analyzer_shared/test/constants/data/errors.dart:9:75 - Methods can't be invoked in constant expressions. - const_eval_method_invocation
```

in https://logs.chromium.org/logs/dart/buildbucket/cr-buildbucket/8756645794309906337/+/u/analyze_pkg_/stdout

Despite the presence of:

```
  exclude:
    - test/constants/data/**
```

**Comment by srawlins** [comment]
@pq Is there progress on this?

**Comment by pq** [comment]
Still simmering... It's a little tricky and needs design work.

**Comment by pq** [comment]
Hopefully this will be fixed w/ https://dart-review.googlesource.com/c/sdk/+/351521.

**Comment by pq** [comment]
https://github.com/dart-lang/sdk/commit/06eea7c263dc3e61c421b1b6aaa2cd586e3bf28b improves matters but we're still not there.

https://logs.chromium.org/logs/dart/buildbucket/cr-buildbucket/8754154758524980737/+/u/analyze_pkg_/stdout

At least 2 issues remain:

1. There are a host of newly reported diagnostics. For example:

`info - vm/testcases/transformations/ffi/finalizable_late_2.dart:19:31 - Statements in a for should be enclosed in a block. Try wrapping the statement in a block. - curly_braces_in_flow_control_structures`

`curly_braces_in_flow_control_structures` should not be enabled here. It's a lint enabled by the `core` set which is included in an options file in the parent directory but without an include here should not be inherited (unless we want to change current semantics).

Here's the relevant options file:

```
analyzer:
  errors:
    sdk_version_since: ignore
  exclude:
    - deferred_loading/**
    - to_string_transformer/**
    - type_flow/**
    - unreachable_code_elimination/**
```

Note no `include:`.

2. Error severity configurations are not working right.

Looking above, you'd expect `sdk_version_since` to be filtered, but it's not:

`warning - vm/testcases/transformations/ffi/ffinative.dart:72:10 - This API is available since SDK 3.3.0, but constraints '>=3.0.0 <4.0.0' don't guarantee it. Try updating the SDK constraints. - sdk_version_since`

**Comment by pq** [comment]
The initial reported issue is fixed; remaining issues tracked in https://github.com/dart-lang/sdk/issues/55252
